### PR TITLE
Add a message when MatchingServerName failed

### PR DESCRIPTION
**Pull request by aaronraimist**
Wording probably could be improved, ~~also it would be nice not to show this for both IPs.~~

Fixes https://github.com/matrix-org/matrix-federation-tester/issues/87 (which should have been in this repo, feel free to transfer)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -83,6 +83,17 @@ let App = create({
                 ERROR: on {ip}: {msg}
               </div>)
           })
+          if (!report.Checks.MatchingServerName) {
```
Comment (richvdh) [inline app.js]: One solution to the duplication-per-IP problem would be to show this if any of the IP addresses have this error:

```javascript
if (Object.values(json.ConnectionReports).some(e=>e.Checks.MatchingServerName)) {
    tldr.push(/* ... */);
}
```

Comment (aaronraimist) [reply app.js]: Yep this works.

File: app.js [hunk 2]
```diff
@@ -83,7 +83,23 @@ let App = create({
...
```
Comment (richvdh) [inline app.js]: redundant `if` statement is redundant

**Comment by richvdh** [comment]
(I've deployed this but cloudflare is caching the old version for now)
